packages/react-async/src/useAsync.spec.js

```diff
@@ -250,4 +250,54 @@ describe("useFetch", () => {
       expect.objectContaining({ preventDefault: expect.any(Function) })
     )
   })
+
+  test("defer=true allows using the promise", () => {
+    let check = 0
+    const component = (
+      <Fetch input="/test" options={{ defer: true }}>
+        {({ run, promise }) => (
+          <button
+            onClick={() => {
+              run()
+              promise
+                .then(() => {
+                  return (check = 1)
+                })
+                .catch(() => {})
+            }}
+          >
+            run
+          </button>
+        )}
+      </Fetch>
+    )
+    const { getByText } = render(component)
+    fireEvent.click(getByText("run"))
+    expect(check).toEqual(1)
+  })
+
+  test("defer=false allows using the promise", () => {
+    let check = 0
+    const component = (
+      <Fetch input="/test" options={{ defer: false }}>
+        {({ run, promise }) => (
+          <button
+            onClick={() => {
+              run()
```

Comment:
Invoking `run()` doesn't make sense here, because `defer: false` means it's automatically invoked on first render.

Reply:
Let's change that so `promise` always exists.

Reply:
You are right, of course. I initially thought, that this only happened on `defer: true` and only tested the non-defered case at the last moment. It's entirely possible that there is no real case for this.